mdtraj/formats/pdb/pdbfile.py

```diff
@@ -704,13 +727,32 @@ def _read_models(self):
                 for j in connect[1:]:
                     if i in atomByNumber and j in atomByNumber:
                         connectBonds.append((atomByNumber[i], atomByNumber[j]))
+
             if len(connectBonds) > 0:
-                # Only add bonds that don't already exist.
-                existingBonds = {(bond.atom1, bond.atom2) for bond in self._topology.bonds}
-                for bond in connectBonds:
-                    if bond not in existingBonds and (bond[1], bond[0]) not in existingBonds:
-                        self._topology.add_bond(bond[0], bond[1])
-                        existingBonds.add(bond)
+                existingBonds = [(bond.atom1, bond.atom2) for bond in self._topology.bonds]
```

Comment:
Is there a reason to keep `existingBonds` in both a list and set version (`existinBonds_set`)? I'm not entirely seeing why the list version is needed.